test/WebJobs.Script.Tests/Extensions/ConfigurationExtensionsTests.cs

```diff
@@ -0,0 +1,159 @@
+            Assert.Equal(true, concurrency["snapshotPersistenceEnabled"]);
+
+            // An integer/boolean value will be transfered to a string value.
+            // HostJosnFileConfigurationSource is also convert these into a string value.
+            Assert.Equal(expectedExtensionsJson, extensionsJson);
+            Assert.Equal(expectedConcurrencyJson, concurrencyJson);
+        }
+
+        [Fact]
+        public void ConvertToJTokenWithEmptyJson_ResultIsPlainObject()
+        {
+            IConfiguration config = BuildConfigurationFromJson("{}");
+            JToken extensions = config.Convert("extensions");
+            Assert.NotNull(extensions);
+            Assert.False(extensions.HasValues);
+        }
+
+        [Fact]
+        public void ConvertToJTokenWithVersionOnly_ResultIsPlainObject()
+        {
+            IConfiguration config = BuildConfigurationFromJson("{\"version\":\"2.0\"}");
+            JToken extensions = config.Convert("extensions");
+            Assert.NotNull(extensions);
+            Assert.False(extensions.HasValues);
+        }
+
+        [Fact]
+        public void ConvertTOJTokenWithMissingSection_ResulthasObjectThatAvaiable()
+        {
+            // Concurrency secrtion is not avaiable but extensions section has it.
+            var hostJson = @"
+{
+  ""version"": ""2.0"",
+  ""extensions"": {
+    ""durableTask"": {
+      ""maxConcurrentActivityFunctions"": 10,
+      ""maxConcurrentOrchestratorFunctions"": 12,
+      ""storageProvider"": {
+        ""type"": ""mssql"",
+        ""connectionStringName"": ""SQLDB_Connection""
+      }
+    }
+  }
+}
+";
+
+            var expectedExtensionsJson = @"{
+  ""durableTask"": {
+    ""maxConcurrentActivityFunctions"": ""10"",
+    ""maxConcurrentOrchestratorFunctions"": ""12"",
+    ""storageProvider"": {
+      ""connectionStringName"": ""SQLDB_Connection"",
+      ""type"": ""mssql""
+    }
+  }
+}";
+            var expectedConcurrencyJson = @"{}";
+
+            IConfiguration config = BuildConfigurationFromJson(hostJson);
+            JToken extensions = config.Convert("extensions");
+            JToken concurrency = config.Convert("concurrency");
+            var extensionsJson = JsonConvert.SerializeObject(extensions, Formatting.Indented);
+            var concurrencyJson = JsonConvert.SerializeObject(concurrency, Formatting.Indented);
+
+            // Actual value is string type, however, it is converted to int.
+            Assert.Equal(10, extensions.SelectToken("durableTask.maxConcurrentActivityFunctions"));
+            Assert.Equal(12, extensions.SelectToken("durableTask.maxConcurrentOrchestratorFunctions"));
+
+            Assert.Null(concurrency["dynamicConcurrencyEnabled"]);
+            Assert.Null(concurrency["snapshotPersistenceEnabled"]);
+
+            // An integer/boolean value will be transfered to a string value.
+            // HostJosnFileConfigurationSource is also convert these into a string value.
```

Comment:
nit: type in comment

Reply:
What do you mean "type in comment"? That is the good comment in this case? I updated the section.

Reply:
Removed this file